### ISSUE TITLE
Problems with first order after enabling DCA

### ISSUE THREAD

**Issue by yoandym** [freqtrade/freqtrade#6268]
## Describe your environment

  * Operating system: ____ Mac
  * Freqtrade Version: ____ docker-730d2e35
  
## Describe the problem:

After setting up a Freqtrade instance in dry mode, with a copy of a live db, and enabling DCA i get:

* Wrong order management
* Wrong average cost calculation
* Wrong profit

### Steps to reproduce:

1. make a copy of a live db, FT stable version, with some open trades
2. setup a dry develop image of FT with the live db copy
3. enable DCA
  
### Observed Results:

  * What happened?
 
1. FT (the strategy) rebuys a coin
2. FT (the strategy) sells the coin (only the amount from last buy) at some price that is not even near the average and declares some incorrect profit (taking in consideration the total amount for the coin)
3. FT closes the whole trade from db (note the amount sold was only the amount from last buy ) 
  

* What did you expect to happen?

2. FT (the strategy) sells the coin (total amount) at some price higher than the average cost and declares the real profit
3. FT closes the whole trade from db only if it completely sells the coin

### Relevant code exceptions or logs

Note: Please copy/paste text of the messages, no screenshots of logs please.

  ```


✓ Binance: Bought WRX/USDT (#66)
Buy Tag: 39 
Amount: 15577.9 
Open Rate: 1.53600000
Total: (23951.68188285 USDT)

✓ Binance: Bought WRX/USDT (#66)
Buy Tag: 39 
Amount: 7590.60000000
Open Rate: 0.94800000
Total: (7195.889 USDT)

  ✳ Binance: Sold WRX/USDT (#66)
Profit: 0.01% (profit: 0.77424120 USDT / 0.774 USD)
Buy Tag: 39 
Sell Reason: sell_profit_t_0_3 ( 39 )
Duration: 49 days, 15:20:18 (71480.3 min)
Amount: 7590.60000000
Open Rate: 0.94800000
Close Rate: 0.95000000
  ```

**Comment by xmatthias** [comment]
Unfortunately, this is the telegram output, not the logs (which is the terminal output freqtrade writes).
We'll need this from the dry-run bot.

also, we'll need the entries from the database for this trade.

```

sqlite3 <databasefile>.sqlite
```

``` sqlite
# Assuming you use sqlite3
.headers on

select * from trades where id = 66;

select * from orders where ft_trade_id = 66;
```

In general, nothing in this regard changed, so updating the bot with open trades to a DCA enabled strategy will not change the result (the other way around might).

**Comment by yoandym** [comment]
attached logs and query response, if more info needed just let me know
note for the first order: in table orders, column average is null/empty maybe that's causing the problem

## logs

```
2022-01-22 16:07:08,230 - freqtrade.persistence.models - INFO - Found open trade: Trade(id=75, pair=PYR/USDT, amount=464.58900000, open_rate=28.00000000, open_since=2021-12-13 00:11:20)

2022-01-22 17:01:02,579 - freqtrade.wallets - INFO - Stake amount for pair PYR/USDT is too big (6575.877700000001 > 2298.758862808427), adjusting to 2298.758862808427.
2022-01-22 17:01:02,581 - freqtrade.freqtradebot - INFO - Position adjust: about to create a new order for PYR/USDT with stake: 2298.758862808427 for Trade(id=75, pair=PYR/USDT, amount=464.58900000, open_rate=28.00000000, open_since=2021-12-13 00:11:20)
2022-01-22 17:01:02,866 - freqtrade.freqtradebot - INFO - Order #dry_run_buy_1642870862.582537 was created for PYR/USDT and status is open.
2022-01-22 17:01:02,902 - freqtrade.wallets - INFO - Wallets synced.
2022-01-22 17:01:02,903 - freqtrade.rpc.rpc_manager - INFO - Sending rpc message: {'trade_id': 75, 'type': buy, 'buy_tag': '18 ', 'exchange': 'Binance', 'pair': 'PYR/USDT', 'limit': 7.99, 'open_rate': 7.99, 'order_type': 'limit', 'stake_amount': 13021.554404565582, 'stake_currency': 'USDT', 'fiat_currency': 'USD', 'amount': 464.589, 'open_date': datetime.datetime(2021, 12, 13, 0, 11, 20, 82932), 'current_rate': 7.99}
2022-01-22 17:01:03,267 - freqtrade.freqtradebot - INFO - DCA order open, will wait for resolution: Trade(id=75, pair=PYR/USDT, amount=464.58900000, open_rate=28.00000000, open_since=2021-12-13 00:11:20)


2022-01-22 17:07:02,141 - freqtrade.freqtradebot - INFO - Found open order for Trade(id=75, pair=PYR/USDT, amount=464.58900000, open_rate=28.00000000, open_since=2021-12-13 00:11:20)
2022-01-22 17:07:02,167 - freqtrade.freqtradebot - INFO - Fee for Trade Trade(id=75, pair=PYR/USDT, amount=464.58900000, open_rate=28.00000000, open_since=2021-12-13 00:11:20) [buy]: 2.298755 USDT - rate: 0.001
2022-01-22 17:07:02,169 - freqtrade.persistence.models - INFO - Updating trade (id=75) ...
2022-01-22 17:07:02,170 - freqtrade.persistence.models - INFO - LIMIT_BUY has been fulfilled for Trade(id=75, pair=PYR/USDT, amount=287.70400000, open_rate=7.99000000, open_since=2021-12-13 00:11:20).
2022-01-22 17:07:02,198 - freqtrade.wallets - INFO - Wallets synced.
2022-01-22 17:07:02,199 - freqtrade.rpc.rpc_manager - INFO - Sending rpc message: {'trade_id': 75, 'type': buy_fill, 'buy_tag': '18 ', 'exchange': 'Binance', 'pair': 'PYR/USDT', 'limit': 7.99, 'open_rate': 7.99, 'order_type': None, 'stake_amount': 2298.75496, 'stake_currency': 'USDT', 'fiat_currency': 'USD', 'amount': 287.704, 'open_date': datetime.datetime(2021, 12, 13, 0, 11, 20, 82932), 'current_rate': 7.99}


2022-01-22 17:10:50,130 - freqtrade.freqtradebot - INFO - Executing Sell for PYR/USDT. Reason: custom_sell. Tag: None
2022-01-22 17:10:50,138 - freqtrade.wallets - INFO - Wallets synced.
2022-01-22 17:10:50,509 - freqtrade.rpc.rpc_manager - INFO - Sending rpc message: {'type': sell, 'trade_id': 75, 'exchange': 'Binance', 'pair': 'PYR/USDT', 'gain': 'profit', 'limit': 8.01, 'order_type': 'limit', 'amount': 287.704, 'open_rate': 7.99, 'close_rate': None, 'current_rate': 8.01, 'profit_amount': 1.150816, 'profit_ratio': 0.00050013, 'buy_tag': '18 ', 'sell_reason': 'sell_profit_t_0_3 ( 18 )', 'open_date': datetime.datetime(2021, 12, 13, 0, 11, 20, 82932), 'close_date': datetime.datetime(2022, 1, 22, 17, 10, 50, 509062), 'stake_currency': 'USDT', 'fiat_currency': 'USD'}
2022-01-22 17:10:50,905 - freqtrade.freqtradebot - INFO - Found open order for Trade(id=75, pair=PYR/USDT, amount=287.70400000, open_rate=7.99000000, open_since=2021-12-13 00:11:20)
2022-01-22 17:10:50,926 - freqtrade.freqtradebot - INFO - Fee for Trade Trade(id=75, pair=PYR/USDT, amount=287.70400000, open_rate=7.99000000, open_since=2021-12-13 00:11:20) [sell]: 2.304509 USDT - rate: 0.001
2022-01-22 17:10:50,927 - freqtrade.persistence.models - INFO - Updating trade (id=75) ...
2022-01-22 17:10:50,928 - freqtrade.persistence.models - INFO - LIMIT_SELL has been fulfilled for Trade(id=75, pair=PYR/USDT, amount=287.70400000, open_rate=7.99000000, open_since=2021-12-13 00:11:20).
2022-01-22 17:10:50,930 - freqtrade.persistence.models - INFO - Marking Trade(id=75, pair=PYR/USDT, amount=287.70400000, open_rate=7.99000000, open_since=closed) as closed as the trade is fulfilled and found no open orders for it.
2022-01-22 17:10:50,962 - freqtrade.wallets - INFO - Wallets synced.
2022-01-22 17:10:50,965 - freqtrade.rpc.rpc_manager - INFO - Sending rpc message: {'type': sell_fill, 'trade_id': 75, 'exchange': 'Binance', 'pair': 'PYR/USDT', 'gain': 'profit', 'limit': 8.01, 'order_type': '', 'amount': 287.704, 'open_rate': 7.99, 'close_rate': 8.01, 'current_rate': None, 'profit_amount': 1.150816, 'profit_ratio': 0.00050013, 'buy_tag': '18 ', 'sell_reason': 'sell_profit_t_0_3 ( 18 )', 'open_date': datetime.datetime(2021, 12, 13, 0, 11, 20, 82932), 'close_date': datetime.datetime(2022, 1, 22, 17, 10, 50, 930272), 'stake_currency': 'USDT', 'fiat_currency': 'USD'}
2022-01-22 17:10:51,948 - freqtrade.wallets - INFO - Wallets synced.
2022-01-22 17:10:53,914 - freqtrade.freqtradebot - INFO - Pair PYR/USDT is still locked until 2022-01-22 17:15:00 due to Auto lock.


```


## select * from trades where id = 75;

```
id|exchange|pair|is_open|fee_open|fee_open_cost|fee_open_currency|fee_close|fee_close_cost|fee_close_currency|open_rate|open_rate_requested|open_trade_value|close_rate|close_rate_requested|close_profit|close_profit_abs|stake_amount|amount|amount_requested|open_date|close_date|open_order_id|stop_loss|stop_loss_pct|initial_stop_loss|initial_stop_loss_pct|stoploss_order_id|stoploss_last_update|max_rate|min_rate|sell_reason|sell_order_status|strategy|buy_tag|timeframe
75|binance|PYR/USDT|0|0.001|2.29875496|USDT|0.001|2.30450904|USDT|7.99|7.99|2301.05371496|8.01|8.01|0.00050013|1.150816|2298.75496|287.704|465.055|2021-12-13 00:11:20.082932|2022-01-22 17:10:50.930272||0.28|-0.99|0.28|-0.99||2021-12-13 00:11:27.326593|28.14|6.74|sell_profit_t_0_3 ( 18 )|closed|NostalgiaForInfinityX|18 |5

```

## select * from orders where ft_trade_id = 75;

```
id|ft_trade_id|ft_order_side|ft_pair|ft_is_open|order_id|status|symbol|order_type|side|price|average|amount|filled|remaining|cost|order_date|order_filled_date|order_update_date
153|75|buy|PYR/USDT|0|10340025|closed|PYR/USDT|limit|buy|28.0||465.055|465.055|0.0|13021.54|2021-12-13 00:11:20.079000|2021-12-13 00:11:25.224123|2021-12-13 00:11:25.224130
170|75|buy|PYR/USDT|0|dry_run_buy_1642870862.582537|closed|PYR/USDT|limit|buy|7.99|7.99|287.704|287.704|0.0|2298.75496|2022-01-22 17:01:02.000000|2022-01-22 17:07:02.144056|2022-01-22 17:07:02.144061
175|75|sell|PYR/USDT|0|dry_run_sell_1642871450.142946|closed|PYR/USDT|limit|sell|8.01|8.01|287.704|287.704|0.0|2304.50904|2022-01-22 17:10:50.000000|2022-01-22 17:10:50.908066|2022-01-22 17:10:50.908072

```

**Comment by xmatthias** [comment]
average is not in filled in 100% of the cases. That's known and we have handling in place for this in almost all areas (obviously we did miss this for DCA orders).

It's now fixed, so with the latest develop, this will no longer happen.
